### PR TITLE
Fixing shadow variables in examples [shadow-examples-dev]

### PR DESCRIPTION
**Pull request by mlstowell**
This PR partially addresses issue #2701.

I tried to make minimal changes and keep the spirit of the original variable names. Hopefully, splitting these into small PRs will assist the original authors and/or reviewers in assessing the changes.

The compiler has no trouble determining which variable is intended but the choice is not always obvious to developers who may not be aware of one or the other of the possibilities.

 | PR | Author | Editor | Reviewers | Assignment | Approval | Merge| 
 | --- | --- | --- | --- | --- | --- | --- | 
| [#2743](https://github.com/mfem/mfem/pull/2743) | @mlstowell | @tzanio | @acfisher + @dylan-copeland | 01/04/22 | 01/18/22 | 01/19/22 | |

### CODE REVIEW

**Comment by tzanio** [comment]
This PR is now [under review](https://github.com/mfem/mfem/blob/master/CONTRIBUTING.md#pull-requests) (see the table in the PR description). To help with the review process, please **do not force push** to the branch.

**Review by tzanio — APPROVED** [review]

File: examples/ex27.cpp [hunk 1]
```diff
@@ -637,11 +638,11 @@ Mesh * GenerateSerialMesh(int ref)
 
 double IntegrateBC(const GridFunction &x, const Array<int> &bdr,
                    double alpha, double beta, double gamma,
-                   double &err)
+                   double &error)
```
Comment (dylan-copeland) [inline examples/ex27.cpp:641]: Line 78 should be changed accordingly.

Comment (mlstowell) [reply examples/ex27.cpp:641]: Fixed

**Review by dylan-copeland — APPROVED** [review]

**Review by acfisher — APPROVED** [review]

**Comment by tzanio** [comment]
@mlstowell, do you plan to change the stream names here too?

**Comment by mlstowell** [comment]
I have not yet started the process of changing the stream names. I can start with this one so we can resolve it.

**Comment by mlstowell** [comment]
@tzanio, the `out_stream -> os` name change is done and the checks have all passed. I think this is ready for the next stage.

**Comment by tzanio** [comment]
thanks @mlstowell

**Comment by tzanio** [comment]
Merged in `next` for testing...

**Pull request merged**